umh-core/pkg/models/action_models.go

```diff
@@ -19,6 +19,35 @@ import (
 	"github.com/google/uuid"
 )
 
+type DeployCustomDataFlowComponentPayload struct {
+	Name              string                         `json:"name" binding:"required"`
+	Payload           DeployDataFlowComponentPayload `json:"payload" binding:"required"`
+	Meta              CdcfMeta                       `json:"meta"`
+	IgnoreHealthCheck bool                           `json:"ignoreHealthCheck"`
+}
+
+type DeployDataFlowComponentPayload struct {
+	CDFCPayload CDFCPayload `json:"customDataFlowComponent" binding:"required"`
+}
+
+type CDFCPayload struct {
+	Input           DfcDataConfig            `json:"input"`
+	Output          DfcDataConfig            `json:"output"`
+	Pipeline        map[string]DfcDataConfig `json:"pipeline"`
+	Inject          DfcDataConfig            `json:"rawYAML"`
+	IgnoreErrors    bool                     `json:"ignoreErrors"`
```

Comment:
this is then for the validation, right? or was it during the deployment? should put a TODO here as this is not working yet